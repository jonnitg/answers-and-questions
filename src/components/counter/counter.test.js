import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Counter from './counter';

afterAll(cleanup);

it('renders counter correctly', () => {
  const { getByText } = render(<Counter limit={20} current={5} />);
  const spanElement = getByText(/question/i);
  expect(spanElement).toHaveTextContent('Question: 5/20');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Counter limit={30} current={3} />).toJSON();
  expect(tree).toMatchSnapshot();
});
