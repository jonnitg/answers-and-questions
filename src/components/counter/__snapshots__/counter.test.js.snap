// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<span
  className="sc-AxjAm khGDyj"
>
  Question: 
  3
  /
  30
</span>
`;
